python/pyspark/sql/tests.py

```diff
@@ -3021,6 +3021,17 @@ def test_sort_with_nulls_order(self):
 
 class HiveSparkSubmitTests(SparkSubmitTests):
 
+    @classmethod
+    def setUpClass(cls):
+        import glob
+        from pyspark.find_spark_home import _find_spark_home
+
+        SPARK_HOME = _find_spark_home()
+        filename_pattern = ("sql/hive/target/spark-hive_*-sources.jar")
+        if not glob.glob(os.path.join(SPARK_HOME, filename_pattern)):
```

Comment:
Since this is test code its probably OK, but this assumes that someone hasn't packaged Spark as an assembly JAR I like the approach taken in https://github.com/apache/spark/pull/20909 for checking.

Reply:
perhaps looks for TestHiveContext like in other test cases
https://github.com/apache/spark/blob/3f1e999d3d215bb3b867bcd83ec5c799448ec730/R/pkg/tests/fulltests/test_sparkSQL.R#L42
https://github.com/apache/spark/blob/61487b308b0169e3108c2ad31674a0c80b8ac5f3/python/pyspark/sql/context.py#L497

Reply:
This PR will follow #20909 like the other occurrence in `HiveContextSQLTests` of this file.